src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -64,20 +63,18 @@ public void getSupportedLocales(MessageCacheItem cacheItem) {
     public List<String> getComponents() {
         List<String> components = new ArrayList<String>();
         try {
-
             Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
-
-            URI uri = Thread.currentThread().getContextClassLoader().
-                    getResource(path.toString()).toURI();
-
-            if (uri.getScheme().equals("jar")) {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-                    path = fileSystem.getPath(path.toString());
+            List<URI> uris = FileUtil.findUris(path);
+            for (URI uri : uris) {
+                if (uri.getScheme().equals("jar")) {
+                    try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                        path = fileSystem.getPath(Paths.get(uri).toString());
+                        getComponents(path, components);
+                    }
+                } else {
+                    path = Paths.get(uri);
                     getComponents(path, components);
                 }
```

Comment:
Here should get the merged result of shared lib's component list and main application's so that all of them can be cached at main application start up, right?

But when the bundle's root path of shared lib and that of main application's is different, you can only get main application's component list.

Reply:
You are right again! This is exactly why we should be able to configure a different offlineResourceBaseURL (and other properties) for shared libs. This is tracked in issue: https://github.com/vmware/singleton/issues/755

Reply:
But I think issue #755 only track the problem that main app and shared libs are able to use different config.

The problem still need to be resolved here is: get the merged result of shared lib's component list and main application's. This problem can be resolved either in PR #858 or in a new PR.

Reply:
You are right. Thats still needs to be resolved after 858. PR 858 is the first step towards that.

For this PR, assume that the main and shared lib's product name and offline bundle paths are exactly the same. There is no problem in a fat executable jar, but there is a problem when using IDE (or whenever the shared lib is a nested jar).